pkg/ebpf/btf.go

```diff
@@ -125,3 +126,47 @@ func loadBTFFrom(path string) (*btf.Spec, error) {
 
 	return btf.LoadSpecFromReader(data)
 }
+
+var kernelSpecCache struct {
+	sync.Mutex
+	refCount int
+	spec     *btf.Spec
+}
+
+// GetKernelSpec returns a possibly cached version of the running kernel BTF spec
+// it's very important that the caller of this function does not modify the returned value
+func GetKernelSpec() (*btf.Spec, error) {
+	kernelSpecCache.Lock()
+	defer kernelSpecCache.Unlock()
+
+	if kernelSpecCache.spec != nil {
+		kernelSpecCache.refCount++
+		return kernelSpecCache.spec, nil
+	}
+
+	// re-sync refCount
+	kernelSpecCache.refCount = 0
+
+	spec, err := btf.LoadKernelSpec()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
shouldn't we cache the error, to prevent "reloading" on unsupported machines?

Reply:
The idea is to fall back to what we were doing before (basically it's not worse in this case). The second point is that `LoadKernelSpec` is not a simple function, it can fail for a multitude of reasons, and I would not like to cache a transient error (IO or similar)